plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal.php

```diff
@@ -1025,10 +1025,6 @@ public function set_transact_onboarding_complete() {
 add_action(
 	'init',
 	function () {
-		if ( ! is_admin() || ( defined( 'DOING_AJAX' ) && DOING_AJAX ) ) {
```

Comment:
Moved the `is_admin` check to the constructor. Removed the doing ajax check as the account restriction action needs to be triggered during checkout.